libs/designer-ui/src/lib/dynamicallyaddedparameter/helper.ts

```diff
@@ -123,20 +123,45 @@ function getDescriptionForDynamicallyAddedParameterType(type: DynamicallyAddedPa
  *              required: ['text', 'text_1'],
  *          }
  *      }
- *
+ * Sample shape of schema object JSON expected by FlowRP that we are serializing/deserializing for Hybrid trigger:
+ *         schema: {
+ *           rows: {
+ *             type: 'array',
+ *             items: {
+ *               type: 'object',
+ *               properties: {
+ *                 type: 'object',
+ *                 properties: {
+ *                   'text' : ...,
+ *                   'text_1': ...,
+ *                   'number': ...,
+ *                 }
+ *                 required: ['text', 'text_1'],
+ *               },
+ *             }
+ *           }
  * @param value - valueSegment provided to us by rest of designer parent components
+ * @param isManualTrigger - Flag that sets the rootObject to required schema type.
  * @param onChange - handler to update value when the user changes their input in one of the dynamic parameters
  * @returns - array of props to render DynamicallyAddedParameter editors with
  */
-export function deserialize(value: ValueSegment[]): DynamicallyAddedParameterProps[] {
+export function deserialize(value: ValueSegment[], isManualTrigger = false): DynamicallyAddedParameterProps[] {
```

Comment:
the parameter is always passed in, there's no need to be optional so I think we should make it required (same with the other helpers)

Reply:
Made it required, with a default false